### PR TITLE
Bugfix/fix chore

### PR DESCRIPTION
**Pull request by nguyenhoaikhang37**
## PR Type:
Bug fix, Refactoring, Enhancement

___
## PR Description:
This PR includes several changes:
- Refactoring of the login form and the review header components.
- Enhancement of the review card component to include a chat with GPT feature.
- Addition of the password input, message list, and message skeleton list components.
- Updates to the environment variables and the services.
- Bug fixes in the search form and the dynamic page.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`app/(auth)/sign-in/_components/login-form.tsx`: Refactored the login form to use the new AppInput and PasswordInput components.
`app/(default)/(private)/reviewing/[...slug]/_components/review-header.tsx`: Refactored the review header to use the Input component from the UI library.
`app/(default)/(private)/reviewing/[...slug]/page.tsx`: Added enhancements to the dynamic page, including updates to the file filtering logic and the addition of a no files found message.
`app/(default)/_components/search-form/index.tsx`: Fixed a bug in the form validation logic.
`app/(default)/reviewing/[...slug]/_components/review-card.tsx`: Enhanced the review card to include a chat with GPT feature.
`components/common/app-input.tsx`: Added a new AppInput component.
`components/common/password-input.tsx`: Added a new PasswordInput component.
`components/common/message-list.tsx`: Added a new MessageList component.
`components/common/message-skeleton-list.tsx`: Added a new MessageSkeletonList component.
`components/dialog/chat-with-gpt-dialog.tsx`: Updated the ChatWithGPTDialog component to use the new MessageList component.
</details>

___
## User Description:
Fix logic dialog and something

### CODE REVIEW

File: app/(default)/_components/search-form/index.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const formSchema = z.object({
       }
       const formattedLink = value.replace('https://github.com/', '').replace('github.com/', '');
 
-      if (formattedLink.split('/').length < 2) {
+      if (formattedLink.split('/').length !== 2) {
```
Comment (tranduybau) [inline app/(default)/_components/search-form/index.tsx]: Lý do bé hơn 2, là vì sẽ có case bằng 3, khi đấy thì vẫn là repos em nha

File: app/(default)/reviewing/[...slug]/_components/review-card.tsx [hunk 2]
```diff
@@ -1,7 +1,9 @@
 'use client';
 
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (tranduybau) [inline app/(default)/reviewing/[...slug]/_components/review-card.tsx]: Sao lại eslint disable rồi @@

File: zustand/useModal/index.ts [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
 // eslint-disable-next-line import/prefer-default-export
-export const useGPTMessageDialog = create<GPTMessageDialog>()(
-  (set) => ({
-    isOpen: false,
-    setIsOpen: (isOpen) => set({ isOpen }),
-  }),
-);
+export const useGPTMessageDialog = create<GPTMessageDialog>((set): GPTMessageDialog => ({
+  selectedFile: '',
+  actionSelectedFileChange: (fileName) => set(() => ({ selectedFile: fileName })),
```
Comment (tranduybau) [inline zustand/useModal/index.ts]:
```suggestion
  actionSelectedFileChange: (fileName) => set(({ selectedFile: fileName })),
```

File: app/(default)/_components/search-form/index.tsx (hunk 1 above)
Comment (nguyenhoaikhang37) [reply app/(default)/_components/search-form/index.tsx]: i've fixed

File: zustand/useModal/index.ts (hunk 3 above)
Comment (nguyenhoaikhang37) [reply zustand/useModal/index.ts]: i've fixed

File: app/(default)/reviewing/[...slug]/_components/review-card.tsx (hunk 2 above)
Comment (nguyenhoaikhang37) [reply app/(default)/reviewing/[...slug]/_components/review-card.tsx]: i've fixed

File: app/(default)/(private)/reviewing/[...slug]/page.tsx [hunk 4]
```diff
@@ -10,17 +10,18 @@ import ChatWithGPTDialog from '@/components/dialog/chat-with-gpt-dialog';
 import { Button } from '@/components/ui/button';
 import { Card, CardContent } from '@/components/ui/card';
 
+import ReviewCard from '@/app/(default)/reviewing/[...slug]/_components/review-card';
```
Comment (tranduybau) [inline app/(default)/(private)/reviewing/[...slug]/page.tsx]:
```suggestion
import ReviewCard from './_components/review-card';
```

**Review by tranduybau — APPROVED** [review]

**Pull request merged**